internal/datastore/memdb/readwrite.go

```diff
@@ -87,6 +97,25 @@ func (rwt *memdbReadWriteTx) write(tx *memdb.Txn, mutations ...*core.RelationTup
 	return nil
 }
 
+func (rwt *memdbReadWriteTx) prepareCaveat(tx *memdb.Txn, mutation *core.RelationTupleUpdate) (*caveatReference, error) {
+	var cr *caveatReference
+	if mutation.Tuple.Caveat != nil {
+		id := datastore.CaveatID(mutation.Tuple.Caveat.CaveatId)
```

Comment:
As discussed, I think this can be done by the calling API layer

Reply:
addressed in https://github.com/authzed/spicedb/pull/807/commits/307d92b61805024e0ea4231308b0db2b127ea375